Libplanet/Blockchain/Policies/VolatileStagePolicy.cs

```diff
@@ -275,5 +250,31 @@ public long GetNextTxNonce(BlockChain<T> blockChain, Address address)
 
         private bool Expired(Transaction<T> transaction) =>
             transaction.Timestamp + Lifetime < DateTimeOffset.UtcNow;
+
+        private Transaction<T>? GetInner(BlockChain<T> blockChain, TxId id, bool filtered)
+        {
+            if (_staged.TryGetValue(id, out Transaction<T>? tx) && tx is { })
+            {
+                if (Expired(tx) || _ignored.Contains(tx.Id))
+                {
+                    _staged.TryRemove(id, out _);
+                    return null;
+                }
+                else if (filtered)
+                {
+                    return blockChain.Store.GetTxNonce(blockChain.Id, tx.Signer) <= tx.Nonce
+                        ? tx
+                        : null;
+                }
+                else
+                {
+                    return tx;
+                }
+            }
+            else
+            {
+                return null;
+            }
+        }
```

Comment:
Not sure why this was separated out. Detaching this from the `lock` statement might make things more prone to mistakes. We might need to at least document it should be operating under a lock?

Reply:
I didn't want unnecessarily lock for inner invocations (e.g., `.Iteratate()` invokes `.Get()` and it already has a read lock).

Reply:
Ah, it is shared with `Iterate()`. I missed that part. 🙃